scripts/ci/pre_commit/pre_commit_generate_airflow_diagrams.py

```diff
@@ -34,201 +30,28 @@
-
-        (
-            auth_manager
-            >> Edge(color="black", style="dotted", reverse=False, label="Inherit")
-            >> auth_manager_interface
-        )
-
-    console.print(f"[green]Generating architecture image {auth_manager_architecture_image_file}")
-
-
-def generate_fab_auth_manager_airflow_diagram():
-    auth_manager_architecture_image_file = (
-        FAB_PROVIDER_DOCS_IMAGES_DIR / FAB_AUTH_MANAGER_AIRFLOW_ARCHITECTURE_IMAGE_NAME
-    ).with_suffix(".png")
-    console.print(f"[bright_blue]Generating architecture image {auth_manager_architecture_image_file}")
-    with Diagram(
-        name="",
-        show=False,
-        direction="LR",
-        curvestyle="ortho",
-        filename=FAB_AUTH_MANAGER_AIRFLOW_ARCHITECTURE_IMAGE_NAME,
-    ):
-        user = User("User")
-        with Cluster("Airflow environment"):
-            webserver = Custom("Webserver(s)", PYTHON_MULTIPROCESS_LOGO.as_posix())
-
-            with Cluster("FAB provider"):
-                fab_auth_manager = Custom("FAB auth manager", PYTHON_MULTIPROCESS_LOGO.as_posix())
-            with Cluster("Core Airflow"):
-                auth_manager_interface = Custom(
-                    "Auth manager\ninterface", PYTHON_MULTIPROCESS_LOGO.as_posix()
-                )
-
-            db = PostgreSQL("Metadata DB")
-
-        user >> Edge(color="black", style="solid", reverse=True, label="Access to the console") >> webserver
-        (
-            webserver
-            >> Edge(color="black", style="solid", reverse=True, label="Is user authorized?")
-            >> fab_auth_manager
-        )
-        (fab_auth_manager >> Edge(color="black", style="solid", reverse=True) >> db)
-        (
-            fab_auth_manager
-            >> Edge(color="black", style="dotted", reverse=False, label="Inherit")
-            >> auth_manager_interface
-        )
-
-    console.print(f"[green]Generating architecture image {auth_manager_architecture_image_file}")
+def _get_file_hash(file_to_check: Path) -> str:
+    hash_md5 = hashlib.md5()
+    hash_md5.update(Path(file_to_check).resolve().read_bytes())
+    return hash_md5.hexdigest()
 
 
 def main():
-    hash_md5 = hashlib.md5()
-    hash_md5.update(Path(__file__).resolve().read_bytes())
-    my_file_hash = hash_md5.hexdigest()
-    hash_file = LOCAL_DIR / DIAGRAM_HASH_FILE_NAME
-    if not hash_file.exists() or not hash_file.read_text().strip() == str(my_file_hash).strip():
-        os.chdir(DOCS_IMAGES_DIR)
-        generate_basic_airflow_diagram()
-        generate_dag_processor_airflow_diagram()
-        generate_auth_manager_airflow_diagram()
-        os.chdir(FAB_PROVIDER_DOCS_IMAGES_DIR)
-        generate_fab_auth_manager_airflow_diagram()
-        os.chdir(DOCS_IMAGES_DIR)
-        hash_file.write_text(str(my_file_hash) + "\n")
-    else:
-        console.print("[bright_blue]No changes to generation script. Not regenerating the images.")
+    for directory in DOCS_IMAGES_DIR, FAB_PROVIDER_DOCS_IMAGES_DIR:
```

Comment:
Would not it be better to set `pass_filenames` to `True` in `.pre-commit-config.yaml` and loop over these files (ignoring md5 files). This way, no need to hardcode the different directories

Reply:
Yeah might be better and simpler, the problem with such approach is that we would have to agree that `diagram_*.py` is the convention that will be universally not used for something else. But I think it's pretty safae to assume this when we limit ourselves to `docs` folder.

Reply:
Done